File: backend/Ecommerce.Domain/src/Shared/QueryOptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Ecommerce.Domain.src.Entities;

namespace Ecommerce.Domain.src.Shared
{
    public class QueryOptions
    {
        public string Search { get; set; } = string.Empty;
        public string Category { get; set; } = "All";
        public string OrderBy { get; set;} = "Newest first";
        public bool OrderByDescending { get; set;} = false;
        public int PageNumber { get; set; } = 0;
        public int PageSize { get; set;} = 30;
    }
}

File: frontend/src/types/QueryOptions.ts
export interface QueryOptions {
    search: string;
    category: string;
    orderBy: string;
    orderByDescending: boolean;
    pageNumber: number;
    pageSize: number;
}

const defaultQuery: QueryOptions = {
    search: "",
    category: "All",
    orderBy: "Newest first",
    orderByDescending: false,
    pageNumber: 0,
    pageSize: 30
}

export default defaultQuery;